### ISSUE TITLE
Support for extension parameters on Azure

### ISSUE THREAD

**Issue by gmagniez** [kubernetes/autoscaler#642]
We use a custom extension (to be precise, a preprovision extension) which needs a parameter, but when trying to scale we face the following error:
``` 
E0209 14:20:37.252154       1 static_autoscaler.go:298] Failed to scale up: failed to increase node group size: resources.DeploymentsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidDeploymentParameterValue" Message="The value of deployment parameter 'register-dnsParameters' is null. Please specify the value or use the parameter reference. See https://aka.ms/arm-deploy/#parameter-file for details."
``` 
It seems that the generated ARM deployment is missing the parameter 'register-dnsParameters' which is defined in our acs-engine cluster definition:
``` 
"extensionProfiles": [
      {
          "name": "register-dns",
          "rootURL": "https://raw.githubusercontent.com/tesharp/acs-engine/register-dns-extension/",
          "version": "v1",
          "script": "register-dns.sh",
          "extensionParameters": "my.domain.name.com"
      }
    ],
```

**Comment by feiskyer** [comment]
@gmagniez Does the extension use securestring parameters? Have you tried `ace-engine scale` command?

**Comment by gmagniez** [comment]
Hello,
I don't think that the extension is using securestrings, It is juste declared like that:
``` 
"preProvisionExtension":
        {
          "name": "register-dns",
          "singleOrAll": "All"
        }
...
"extensionProfiles": [
      {
          "name": "register-dns",
          "rootURL": "https://raw.githubusercontent.com/tesharp/acs-engine/register-dns-extension/",
          "version": "v1",
          "script": "register-dns.sh",
          "extensionParameters": "my.domain.name.com"
      }
    ],
```

Scaling with 'acs-engine scale' works perfectly.

**Comment by feiskyer** [comment]
@gmagniez Thanks, I will check how to make this work.

**Comment by feiskyer** [comment]
Fixed in #682
